java/client/src/main/java/glide/managers/CommandManager.java

```diff
@@ -29,83 +29,203 @@ public class CommandManager {
     /**
      * Build a command and send.
      *
-     * @param command The command to execute
+     * @param requestType Redis command type
+     * @param arguments Redis command arguments
+     * @param route Command routing parameters
      * @param responseHandler The handler for the response object
      * @return A result promise of type T
      */
     public <T> CompletableFuture<T> submitNewCommand(
-            Command command, RedisExceptionCheckedFunction<Response, T> responseHandler) {
-        // write command request to channel
-        // when complete, convert the response to our expected type T using the given responseHandler
-        return channel
-                .write(
-                        prepareRedisRequest(
-                                command.getRequestType(),
-                                command.getArguments(),
-                                Optional.ofNullable(command.getRoute())),
-                        true)
-                .thenApplyAsync(responseHandler::apply);
+            RequestType requestType,
+            String[] arguments,
+            Optional<RequestRoutingConfiguration.Route> route,
```

Comment:
Please update import and
```suggestion
            Optional<Route> route,
```

Reply:
Fixed